packages/story-editor/src/app/rightClickMenu/hooks/useCopyPasteActions.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { noop, useSnackbar } from '@googleforcreators/design-system';
+import { __ } from '@googleforcreators/i18n';
+import { useCallback, useRef } from '@googleforcreators/react';
+import { trackEvent } from '@googleforcreators/tracking';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import { PRESET_TYPES } from '../../../constants';
+import updateProperties from '../../../components/inspector/design/updateProperties';
+import useApplyStyle from '../../../components/panels/design/textStyle/stylePresets/useApplyStyle';
+import getUpdatedSizeAndPosition from '../../../utils/getUpdatedSizeAndPosition';
+import { getTextPresets } from '../../../utils/presetUtils';
+import { useHistory, useStory } from '../..';
+import { ACTION_TYPES } from '../reducer';
+import { getElementStyles } from './utils';
+import { UNDO_HELP_TEXT } from './constants';
+
+/**
+ * Creates the right click menu copy/paste actions.
+ *
+ * @param {Function} dispatch The right click reducer dispatch function
+ * @param {Object} copiedElement Stored element styles
+ * @return {Object} Right click menu copy/paste actions
+ */
+const useCopyPasteActions = (dispatch = noop, copiedElement) => {
```

Comment:
💭 Don't love having to pass in the dispatch function and our stored state from the reducer. However, the reducer is initialized in the provider. Do y'all have any ideas of better things to do here?

Making an entire new context for one function and this state seems like overkill.

Reply:
looking at `ACTION_TYPES.COPY_ELEMENT_STYLES` & `copiedElement`, I'm not seeing why they're needed to be stored in the reducer state.

Would it be cleaner just to `const [copiedElement, setCopiedElement] = useState(null)` internally here? and return `copiedElement` along with `onCopyStyles` & `onPasteStyles` from this hook?

Reply:
Yeah, there's no need for that state to be included in the reducer state. Just how it was set up initially

Good catch - it would be better to move that state into this hook.